src/react/databrowser/objects/ObjectListTable.jsx

```diff
@@ -83,11 +83,22 @@ export default function ObjectListTable({
   const query = useQuery();
   const objectKey = query.get('prefix');
   const versionId = query.get('versionId');
+  const isListVersion = query.get('showversions') === 'true';
 
   useEffect(() => {
-    dispatch(toggleObject(objectKey, versionId ? versionId : undefined));
-  }, [dispatch]);
+    dispatch(
+      toggleObject(
+        objectKey,
+        versionId && isListVersion ? versionId : undefined,
+      ),
+    );
+    // Add `objectsLength` as dependency to make sure the toogleObject action is triggered after listObject
```

Comment:
Then I guess we should wrap the above line with an `if (objectsLength) {}` so that it is more explicit and we don't need this comment to explain why we need that dependency

Reply:
Good point !!